lib/sparkql/parser_tools.rb

```diff
@@ -43,6 +43,7 @@ def tokenize_expression(field, op, val)
     expression = {:field => field, :operator => operator, :conjunction => 'And', 
       :level => @lexer.level, :block_group => block_group, :custom_field => custom_field}.merge!(field_args)
     expression = val.merge(expression) unless val.nil?
+    expression[:condition] ||= expression[:value]
```

Comment:
This assumes that if the condition wasn't set by now using the function or list builder, it much just be a literal value. As an alternative, I could have built a new intermediary step, but that seems overkill.